sympy/integrals/laplace.py

```diff
@@ -1009,30 +1008,21 @@ def _piecewise_to_heaviside(f, t):
     """
     x = piecewise_exclusive(f)
     r = S.Zero
-    for term in x.args:
+    for fn, cond in x.args:
         # Here we do not need to do many checks because piecewise_exclusive
         # has a clearly predictable output. However, if any of the conditions
         # is not relative to t, this function just returns the input argument.
-        fn = term[0]
-        op = term[1].func
-        if op == LessThan or op == StrictLessThan:
-            if not term[1].args[0] == t:
-                return f
-            right_boundary = term[1].args[1]
-            r = r + Heaviside(-(t-right_boundary))*fn
-        if op == GreaterThan or op == StrictGreaterThan:
-            if not term[1].args[0] == t:
-                return f
-            left_boundary = term[1].args[1]
-            r = r + Heaviside(t-left_boundary)*fn
-        if op == And:
-            if not (
-                    term[1].args[0].args[0] == t and
-                    term[1].args[1].args[0] == t):
-                return f
-            left_boundary = term[1].args[0].args[1]
-            right_boundary = term[1].args[1].args[1]
-            r = r + (Heaviside(t-right_boundary)-Heaviside(t-left_boundary))*fn
+        if isinstance(cond, Relational) and t in cond.args:
+            r = r + Heaviside(cond.gts - cond.lts)*fn
+        elif (    # And(t <= a, t > b)
```

Comment:
It might come back as `And(t >= b, t < a)`, too:

![image](https://github.com/sympy/sympy/assets/90703/730c8eb5-4ae0-4087-9255-4a0ed807f1cf)

Reply:
Unless this can never happen, you should add a test for this case and then consider the modifications I show

Reply:
Or better, update piecewise exclusive so it is consistent for that first case.